tfjs-backend-webgpu/src/kernels/depth_to_space_webgpu.ts

```diff
@@ -0,0 +1,141 @@
+  size: number;
+  useWgsl: boolean;
+
+  constructor(
+      outputShape: number[], blockSize: number, dataFormat: 'NHWC'|'NCHW') {
+    this.outputShape = outputShape;
+    this.dispatchLayout = flatDispatchLayout(this.outputShape);
+    this.dispatch = computeDispatch(
+        this.dispatchLayout, this.outputShape, this.workGroupSize);
+    this.shaderKey = `depthToSpace_${outputShape}_${blockSize}_${dataFormat}`;
+    this.size = util.sizeFromShape(this.outputShape);
+    this.blockSize = blockSize;
+    this.dataFormat = dataFormat;
+  }
+
+  getUserCode(): string {
+    const userCode = `
+      void main() {
+        int index = getGlobalIndex();
+        if (index < size) {
+          ivec4 coords = getOutputCoords();
+          int b = coords[0];
+          int h = ${this.getHeightCoordString()};
+          int w = ${this.getWidthCoordString()};
+          int d = ${this.getDepthCoordString()};
+
+          int in_h = h / ${this.blockSize};
+          int offset_h = h % ${this.blockSize};
+          int in_w = w / ${this.blockSize};
+          int offset_w = w % ${this.blockSize};
+          int offset_d = (offset_h * ${this.blockSize} + offset_w) *
+            ${this.getOutputDepthSize()};
+          int in_d = d + offset_d;
+
+          float result = ${this.getInputSamplingString()};
+          setOutput(index, result);
+        }
+      }`;
+    return userCode;
+  }
+
+  getUserCodeWgsl(): string {
+    const userCode = `
+      ${getMainHeaderStringWgsl(this.workGroupSize)} {
+        ${getGlobalIndexStringWgsl(this.workGroupSize)}
+        let index = getGlobalIndex();
+        if (index < size) {
+          let coords = getOutputCoords();
+          let b = coords[0];
+          let h = ${this.getHeightCoordString()};
+          let w = ${this.getWidthCoordString()};
+          let d = ${this.getDepthCoordString()};
+
+          let in_h = h / ${this.blockSize};
+          let offset_h = h % ${this.blockSize};
+          let in_w = w / ${this.blockSize};
+          let offset_w = w % ${this.blockSize};
+          let offset_d = (offset_h * ${this.blockSize} + offset_w) *
+            ${this.getOutputDepthSize()};
+          let in_d = d + offset_d;
+
+          let result = ${this.getInputSamplingString()};
+          setOutput(index, result);
+        }
+      }`;
+    return userCode;
+  }
+
+  private getHeightCoordString(): string {
+    if (this.dataFormat === 'NHWC') {
+      return `coords[1]`;
+    } else {
+      return `coords[2]`;
+    }
+  }
+
+  private getWidthCoordString(): string {
+    if (this.dataFormat === 'NHWC') {
+      return `coords[2]`;
+    } else {
+      return `coords[3]`;
+    }
+  }
+
+  private getDepthCoordString(): string {
+    if (this.dataFormat === 'NHWC') {
+      return `coords[3]`;
+    } else {
+      return `coords[1]`;
+    }
+  }
+
+  private getOutputDepthSize(): number {
+    if (this.dataFormat === 'NHWC') {
+      return this.outputShape[3];
```

Comment:
Please use the uniform `outShape[3]` instead of the constant value so that you can remove `${outputShape}` from shader key.

Reply:
Done.